src/Features/Core/Portable/ExtractMethod/MethodExtractor.AnalyzerResult.cs

```diff
@@ -67,13 +69,8 @@ protected sealed class AnalyzerResult(
 
             public ImmutableArray<VariableInfo> Variables { get; } = variables;
 
-            public bool HasReturnType
-            {
-                get
-                {
-                    return ReturnType.SpecialType != SpecialType.System_Void && !AwaitTaskReturn;
```

Comment:
note: i genuinely don't know what this expression meant before.  especially the AwaitTaskReturn part.  This is much clearer now.  There is a core return type (prior to any task wrapping) and it can be checked.  To know if there is a return type or not, it can be checked against System.Void.